common/object-association.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+// For now this only has per-Window tests, but we could expand it to also test per-Document
+
+window.testIsPerWindow = propertyName => {
+  test(t => {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    const frame = iframe.contentWindow;
+
+    const before = frame[propertyName];
+    assert_true(before !== undefined && before !== null, `window.${propertyName} must be implemented`);
+
+    iframe.remove();
+
+    const after = frame[propertyName];
+    assert_equals(after, before);
+  }, `Discarding the browsing context must not change window.${propertyName}`);
+
+  async_test(t => {
+    const iframe = document.createElement("iframe");
+    document.body.appendChild(iframe);
+    const frame = iframe.contentWindow;
+
+    const before = frame[propertyName];
+    assert_true(before !== undefined && before !== null, `window.${propertyName} must be implemented`);
+
+    iframe.onload = t.step_func(() => {
+      if (frame.location.href === "about:blank") {
+        // Browsers are not reliable on whether about:blank fires the load event; see
+        // https://github.com/whatwg/html/issues/490
+        return;
+      }
+
+      const after = frame[propertyName];
+      assert_equals(after, before);
+      t.done();
```

Comment:
Cannot be used here because we don't want to be done if the location is about:blank. I'll add a comment.